pandora-common/src/services/serviceManager.ts

```diff
@@ -0,0 +1,4 @@
+export type Service = {
+	init?(): Promise<Service> | Service;
+	onDestroy?(): Promise<void> | void;
```

Comment:
Nit: Maybe `destroy` would be better? Having an "on" prefix signifies that you should never call this method yourself - it happens as response to some event... Hmm~ I guess I feel neutral, as it kinda is reaction to server being destroyed... but maybe then `onShutdown` might suit better?

Reply:
well we did have onDestroy for most of the cases, so that's why I went with it